WordPress/openverse
# Use a modal for license explanation

## Fixes



Fixes #3582 by @fcoveram
Fixes #4802 by @fcoveram 

## Description


This PR converts the license explanation popover to a modal.

### Desktop

<img width="400" alt="Screenshot 2024-08-23 at 11 43 00 AM" src="https://github.com/user-attachments/assets/11521c74-a814-4014-aa85-a3f884e071d2">

### Mobile
<img width="400" alt="Screenshot 2024-08-23 at 11 43 41 AM" src="https://github.com/user-attachments/assets/4bbb09f4-0cdd-46ed-a708-33fd637578d4">

### Functionality changes
On mobile screens, the license explanation modal appears on top of the content settings modal. So, when the user clicked outside of the license explanation modal, the lower content settings modal handled this event and closed, keeping the top-most license explanation one open.
To correctly handle this, I added a `use-dialog-stack` composable and `id`s to all modals and popovers to keep track and route the event to the currently open, top-most dialog.

### Vertical overflow
I added the `max-height` to modals, set to the full height of the screen, minus the height of the modal header, minus an arbitrary value of 6rem. @fcoveram, what is the best value for vertical margin of the modal?

<img width="400" alt="Screenshot 2024-08-23 at 11 47 30 AM" src="https://github.com/user-attachments/assets/35b9742d-57d0-4785-9a64-7fd631abbf31">

## Testing Instructions

Run the app and go to `http://localhost:8443/search?q=cat`.
On desktop, open the filters sidebar and click on one of the `?` buttons next to licenses. The modal should appear. It's styles should match the mockups, and it should function correctly.
On a mobile, open the content settings modal, select the _Filters_ tab and click on one of the `?` buttons next to licenses. The modal should appear. Clicking on the close button, pressing <kbd>Escape</kbd> or clicking outside of the modal should close it, and only it (not the content settings modal below it).

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
